apis/python/src/tiledbsoma/sparse_nd_array.py

```diff
@@ -374,3 +301,68 @@ def tables(self) -> TableReadIter:
         [lifecycle: experimental]
         """
         return TableReadIter(self.sr)
+
+
+def _build_tiledb_schema(
+    type: pa.DataType,
+    shape: Sequence[int],
+    create_options: TileDBCreateOptions,
+    context: SOMATileDBContext,
+    *,
+    is_sparse: bool,
+) -> tiledb.ArraySchema:
+    util.check_type("type", type, (pa.DataType,))
+
+    # check on shape
+    if len(shape) == 0 or any(e <= 0 for e in shape):
+        raise ValueError("DenseNDArray shape must be non-zero length tuple of ints > 0")
+
+    if not pa.types.is_primitive(type):
+        raise TypeError(
+            f"Unsupported type {type} --"
+            " SOMA NDArrays only support primtive Arrow types"
+        )
+
+    dims = []
+    for n, e in enumerate(shape):
+        dim_name = f"soma_dim_{n}"
+        dim = tiledb.Dim(
+            name=dim_name,
+            domain=(0, e - 1),
+            tile=create_options.dim_tile(dim_name, min(e, 2048)),
```

Comment:
this is missing a fix John just applied to main in #822

Reply:
Picked this up upon rebasing.